ex03.ex

```diff
@@ -54,8 +54,11 @@ defmodule Ex03 do
   be used if needed.)
 
   """
+  def odd_even([]) do [] end
```

Comment:
## -2

You can take pattern matching a level deeper here and eliminate the `if`.

~~~ elixir
  def odd_even([]), do: []
  def odd_even([ h | t ]), do: [ atom_for(h) | odd_even(t) ]

  defp atom_for(n) when Integer.is_odd(n),  do:  :odd
  defp atom_for(n) when Integer.is_even(n), do:  :even
~~~

This also eliminates the bug in your version (what happens if you pass it `[ 1, "cat", 3, :wombat ]`?